posthog/warehouse/api/saved_query.py

```diff
@@ -405,6 +409,38 @@ def run(self, request: request.Request, *args, **kwargs) -> response.Response:
 
         return response.Response(status=status.HTTP_200_OK)
 
+    @action(methods=["POST"], detail=True)
+    def revert_materialization(self, request: request.Request, *args, **kwargs) -> response.Response:
+        """
+        Undo materialization, revert back to the original view.
+        (i.e. delete the materialized table and the schedule)
+        """
+        saved_query = self.get_object()
+
+        with transaction.atomic():
+            saved_query.sync_frequency_interval = None
+            saved_query.last_run_at = None
+            saved_query.latest_error = None
+            saved_query.status = None
+
+            # delete the materialized table reference
+            if saved_query.table is not None:
+                saved_query.table.soft_delete()
+                saved_query.table_id = None
+
+            try:
+                delete_saved_query_schedule(str(saved_query.id))
+            except Exception as e:
+                logger.exception(f"Failed to delete temporal schedule for saved query {saved_query.id}: {str(e)}")
+
+            saved_query.save()
+
+            DataWarehouseModelPath.objects.filter(
+                team=saved_query.team, path__lquery=f"*{{1,}}.{saved_query.id.hex}"
+            ).delete()
```

Comment:
👍 out of scope for this PR but noting here I think on updates, we should similarly be clearing and rebuilding the paths